Author: Laurent Bigonville <bigon@debian.org
Description:
 Handle 'nosplash' boot parameter as 'plymouth.enable=0' in system unit

--- a/systemd-units/plymouth-halt.service.in
+++ b/systemd-units/plymouth-halt.service.in
@@ -4,6 +4,7 @@ After=getty@tty1.service display-manager
 Before=systemd-halt.service
 DefaultDependencies=no
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 
 [Service]
--- a/systemd-units/plymouth-kexec.service.in
+++ b/systemd-units/plymouth-kexec.service.in
@@ -4,6 +4,7 @@ After=getty@tty1.service display-manager
 Before=systemd-kexec.service
 DefaultDependencies=no
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 
 [Service]
--- a/systemd-units/plymouth-poweroff.service.in
+++ b/systemd-units/plymouth-poweroff.service.in
@@ -4,6 +4,7 @@ After=getty@tty1.service display-manager
 Before=systemd-poweroff.service
 DefaultDependencies=no
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 
 [Service]
--- a/systemd-units/plymouth-reboot.service.in
+++ b/systemd-units/plymouth-reboot.service.in
@@ -4,6 +4,7 @@ After=getty@tty1.service display-manager
 Before=systemd-reboot.service
 DefaultDependencies=no
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 
 [Service]
--- a/systemd-units/plymouth-start.service.in
+++ b/systemd-units/plymouth-start.service.in
@@ -5,6 +5,7 @@ Wants=systemd-ask-password-plymouth.path
 After=systemd-vconsole-setup.service systemd-udev-trigger.service systemd-udevd.service
 Before=systemd-ask-password-plymouth.service
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 IgnoreOnIsolate=true
 
--- a/systemd-units/systemd-ask-password-plymouth.path.in
+++ b/systemd-units/systemd-ask-password-plymouth.path.in
@@ -6,6 +6,7 @@ Conflicts=shutdown.target
 After=plymouth-start.service
 Before=basic.target shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionPathExists=/run/plymouth/pid
 ConditionVirtualization=!container
 
--- a/systemd-units/systemd-ask-password-plymouth.service.in
+++ b/systemd-units/systemd-ask-password-plymouth.service.in
@@ -6,6 +6,7 @@ Conflicts=shutdown.target
 After=plymouth-start.service
 Before=shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
+ConditionKernelCommandLine=!nosplash
 ConditionVirtualization=!container
 ConditionPathExists=@plymouthruntimedir@/pid
 
